pkg/deployments/tasks/20230206-erc4626-linear-pool-v3/test/task.fork.ts

```diff
@@ -0,0 +1,404 @@
+      const scaledCash = cash.mul(FRXETH_SCALING);
+
+      let fees;
+      if (scaledCash.gt(upperTarget)) {
+        expect(expectedState).to.equal(LinearPoolState.MAIN_EXCESS);
+
+        const excess = scaledCash.sub(upperTarget);
+        fees = excess.mul(SWAP_FEE_PERCENTAGE).div(FP_ONE);
+      } else if (scaledCash.lt(lowerTarget)) {
+        expect(expectedState).to.equal(LinearPoolState.MAIN_LACK);
+
+        const lack = lowerTarget.sub(scaledCash);
+        fees = lack.mul(SWAP_FEE_PERCENTAGE).div(FP_ONE);
+      } else {
+        expect(expectedState).to.equal(LinearPoolState.BALANCED);
+
+        fees = 0;
+      }
+
+      const initialRecipientMainBalance = await mainToken.balanceOf(other.address);
+      if (expectedState != LinearPoolState.BALANCED) {
+        await rebalancer.connect(holder).rebalance(other.address);
+      } else {
+        await rebalancer.connect(holder).rebalanceWithExtraMain(other.address, 5);
+      }
+      const finalRecipientMainBalance = await mainToken.balanceOf(other.address);
+
+      if (fees > 0) {
+        // The recipient of the rebalance call should get the fees that were collected (though there's some rounding
+        // error in the main-wrapped conversion).
+        expect(finalRecipientMainBalance.sub(initialRecipientMainBalance)).to.be.almostEqual(
+          fees.div(FRXETH_SCALING),
+          0.00000001
+        );
+      } else {
+        // The recipient of the rebalance call will get any extra main tokens that were not utilized.
+        expect(finalRecipientMainBalance).to.be.almostEqual(initialRecipientMainBalance, 0.00000001);
+      }
+
+      const mainInfo = await vault.getPoolTokenInfo(poolId, frxEth);
+
+      const expectedMainBalance = lowerTarget.add(upperTarget).div(2);
+      expect(mainInfo.cash.mul(FRXETH_SCALING)).to.equal(expectedMainBalance);
+      expect(mainInfo.managed).to.equal(0);
+    });
+  }
+
+  describe('create and check getters', () => {
+    it('deploy a linear pool', async () => {
+      const tx = await factory.create(
+        '',
+        '',
+        frxEth,
+        erc4626Token,
+        INITIAL_UPPER_TARGET,
+        SWAP_FEE_PERCENTAGE,
+        owner.address,
+        PROTOCOL_ID
+      );
+      const event = expectEvent.inReceipt(await tx.wait(), 'PoolCreated');
+
+      pool = await task.instanceAt('ERC4626LinearPool', event.args.pool);
+      expect(await factory.isPoolFromFactory(pool.address)).to.be.true;
+
+      poolId = await pool.getPoolId();
+      const [registeredAddress] = await vault.getPool(poolId);
+      expect(registeredAddress).to.equal(pool.address);
+
+      const { assetManager } = await vault.getPoolTokenInfo(poolId, frxEth); // We could query for either frxEth or erc4626Token
+      rebalancer = await task.instanceAt('ERC4626LinearPoolRebalancer', assetManager);
+
+      await mainToken.connect(holder).approve(rebalancer.address, MAX_UINT256); // To send extra main on rebalance
```

Comment:
Maybe consider ending the test in the line that checks the address?
```
expect(registeredAddress).to.equal(pool.address);
```
The approval seems to be part of a subsequent test.